src/miral/render_scene_into_surface.cpp

```diff
@@ -95,53 +95,41 @@ class BufferPool
         std::lock_guard lock{mutex};
         for (size_t i = 0; i < buffers.size(); ++i)
         {
-            if (!claimed[i])
+            auto& pair = buffers[i];
+            if (!pair.first)
```

Comment:
Tracking what `pair.first` and `pair.second` mean isn't easy on the reader. How about:
 
```suggestion
            if (auto& [used, buffer] = buffers[i]; !used)
```